### ISSUE TITLE
Bug: no-mixed-operators rule not catching violations

### ISSUE THREAD

**Issue by KurtPreston** [eslint/eslint#15870]
### Environment

Node version: 14.18.2
npm version: 6.14.15
Local ESLint version: 8.15.0
Global ESLint version: n/a
Operating System: Linux

### What parser are you using?

@typescript-eslint/parser

### What did you do?

I am trying to use `no-mixed-operators` to catch cases where the `if` clause in a ternary is ambiguous. There's an example provided [in the no-mixed-operators rule documentation](https://eslint.org/docs/rules/no-mixed-operators) that shows exactly this:

```
var foo = a && b ? c : d;      /*BAD: Unexpected mix of '&&' and '?:'.*/
```

However, eslint does not report any lint errors on this example.

**Configuration:**
```
{
  rules: {
    'no-mixed-operators': 'error',
  }
};
```

**Code:**
```js
const a = true;
const b = false;
const c = 5;
const d = 6;
const foo = a && b ? c : d;
```

### What did you expect to happen?

Lint error 

### What actually happened?

No error found

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Link to example:
https://eslint.org/demo#eyJ0ZXh0IjoiLy8gU2V0LXVwIHZhcmlhYmxlc1xuY29uc3QgYSA9IHRydWU7XG5jb25zdCBiID0gZmFsc2U7XG5jb25zdCBjID0gNTtcbmNvbnN0IGQgPSA2O1xuXG4vLyBDb3B5IGV4YW1wbGVzIGZyb20gdGhlIHJ1bGUgZG9jdW1lbnRhdGlvblxudmFyIGZvbyA9IGEgJiYgYiB8fCBjIHx8IGQ7ICAgIC8qQkFEKi9cbnZhciBmb28gPSBhICYmIGIgPyBjIDogZDsgICAgICAvKlN1cHBvc2VkIHRvIGJlIEJBRCwgYnV0IHJlcG9ydGluZyBHT09EKi9cbnZhciBmb28gPSAoYSAmJiBiKSA/IGMgOiBkOyAgICAvKkdPT0QqL1xudmFyIGZvbyA9IChhICYmIGIpIHx8IGMgfHwgZDsgIC8qR09PRCovXG52YXIgZm9vID0gYSAmJiAoYiB8fCBjIHx8IGQpOyAgLypHT09EKi8iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7Im5vLWRlYnVnZ2VyIjoyLCJuby1kZWxldGUtdmFyIjoyLCJuby1kdXBlLWFyZ3MiOjIsIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6Miwibm8tZHVwZS1lbHNlLWlmIjoyLCJuby1kdXBlLWtleXMiOjIsIm5vLWR1cGxpY2F0ZS1jYXNlIjoyLCJuby1lbXB0eSI6Miwibm8tZW1wdHktY2hhcmFjdGVyLWNsYXNzIjoyLCJuby1lbXB0eS1wYXR0ZXJuIjoyLCJuby1leC1hc3NpZ24iOjIsIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6Miwibm8tZXh0cmEtc2VtaSI6Miwibm8tZmFsbHRocm91Z2giOjIsIm5vLWZ1bmMtYXNzaWduIjoyLCJuby10aGlzLWJlZm9yZS1zdXBlciI6Miwibm8tdW5kZWYiOjIsIm5vLXVuZXhwZWN0ZWQtbXVsdGlsaW5lIjoyLCJuby11bnJlYWNoYWJsZSI6Miwibm8tdW5zYWZlLWZpbmFsbHkiOjIsIm5vLXVuc2FmZS1uZWdhdGlvbiI6Miwibm8tdW5zYWZlLW9wdGlvbmFsLWNoYWluaW5nIjoyLCJuby11bnVzZWQtbGFiZWxzIjoyLCJuby11c2VsZXNzLWJhY2tyZWZlcmVuY2UiOjIsIm5vLXVzZWxlc3MtY2F0Y2giOjIsIm5vLXVzZWxlc3MtZXNjYXBlIjoyLCJuby13aXRoIjoyLCJyZXF1aXJlLXlpZWxkIjoyLCJ1c2UtaXNuYW4iOjIsInZhbGlkLXR5cGVvZiI6Miwibm8tbWl4ZWQtb3BlcmF0b3JzIjoyfSwiZW52Ijp7fX19

**Comment by mdjermanovic** [comment]
Hi @KurtPreston, thanks for the issue!

 

> **Code:**
> 
> ```js
> const a = true;
> const b = false;
> const c = 5;
> const d = 6;
> const foo = a && b || c || d;
> ```

To clarify, when configured as `'no-mixed-operators': 'error'`, this rule reports a lint error on `a && b || c || d`, but you are also expecting a lint error on `a && b ? c : d` which isn't reported?

**Comment by KurtPreston** [comment]
@mdjermanovic Correct. Sorry for confusion, I realized I copied the wrong line in my report. Updating the ticket description now.

```
// eslint correctly reports a lint error on this line
var foo = a && b || c || d;

// eslint *should* report a lint error on this line, but does not
var foo = a && b ? c : d;
```

**Comment by mdjermanovic** [comment]
This works as intended because this rule reports only operators that are in the same group.

Default groups are as in the configuration example provided at the beginning of the [Options](https://eslint.org/docs/rules/no-mixed-operators#options) section.

For example:

```js
/* eslint no-mixed-operators: "error" */

// "groups" is not configured, so the default groups apply

a + b * c; // error, because "+" and "*" are in the same group

a && b || c ; // error, because "&&" and "||" are in the same group

a + b || c; // no error, because "+" and "||" are not in the same group
```

Neither of the default groups includes ternary operator, so by default it can be mixed with any other: 

> Note: Ternary operator(?:) can be part of any group and by default is allowed to be mixed with other operators.

You can reconfigure groups as you like, for example:

```js
/*eslint no-mixed-operators: ["error", { groups: [
    ["&&", "||", "?:"]
]}]*/

const a = true;
const b = false;
const c = 5;
const d = 6;
const foo = a && b ? c : d; // error: Unexpected mix of '&&' and '?:'.
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tbWl4ZWQtb3BlcmF0b3JzOiBbXCJlcnJvclwiLCB7IGdyb3VwczogW1xuICAgIFtcIiYmXCIsIFwifHxcIiwgXCI/OlwiXVxuXX1dKi9cblxuY29uc3QgYSA9IHRydWU7XG5jb25zdCBiID0gZmFsc2U7XG5jb25zdCBjID0gNTtcbmNvbnN0IGQgPSA2O1xuY29uc3QgZm9vID0gYSAmJiBiID8gYyA6IGQ7IC8vIGVycm9yOiBVbmV4cGVjdGVkIG1peCBvZiAnJiYnIGFuZCAnPzonLiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

> I am trying to use `no-mixed-operators` to catch cases where the `if` clause in a ternary is ambiguous. There's an example provided [in the no-mixed-operators rule documentation](https://eslint.org/docs/rules/no-mixed-operators) that shows exactly this:
> 
> ```
> var foo = a && b ? c : d;      /*BAD: Unexpected mix of '&&' and '?:'.*/
> ```

The rule works as intended (per the default configuration), but the documentation does seem confusing because the opening section uses some examples that are not errors by default, so maybe we should remove them.
